Support Moodle and 2FA

FurTactics opened the pull request:
Hi
I have extended your script due to the introduction of two-factor authentication system and now it has two versions:

1. It allows you to enter a six digit OTP code on the keyboard, or the [2FA extension](https://2fas.com/browser-extension/) does it
2. It generates the code itself using a secret key, which can be obtained by decrypting the QR code

Also, I often use the [e-sprachen Moodle](https://tud.uni-leipzig.de/moodle2/) service, so I added an autologin for it as well.
I have a question about the possibility of copyrights and distribution. I'm just starting my way into GitHub and coding and I'm not quite sure what the best thing to do (or the honest and correct thing to do) is: who will distribute the new version of the code? You or me? Or can we do it together?
Anyway, thanks for your development. It helped me a lot!

Best wishes
FurTactics

FurTactics committed:
Support 2FA and Moodle

The ability to auto-login in Moodle (foreign language learning platform from LSK MSZ) has been added. The code also waits for a two-factor authentication code to be entered (e.g. as recommended by TUD via the 2FA extension or on the keyboard)

FurTactics committed:
TUD Autologin with generating the 2FA

The ability to auto-login in Moodle (foreign language learning platform from LSK MSZ) has been added. The script also generates a six-digit OTP code by a given key (ONLY FOR GEEKS)

FurTactics committed:
Bug Fixes and no infinite loops

FurTactics committed:
Update instructions

FurTactics committed:
Update TUD AutoLogin script with improved descriptions, support for additional platforms and 2FA(S)

OliEfr commented:
Hey, there is also [www.tu-fast.de](www.tu-fast.de), which currently supports 2FA, maybe that helps :)

Disclaimer: I'm a Dev

FurTactics commented:
Hey @OliEfr, yes I know. It's also an interesting project, but I need something with less space and more configurable (I need support for e.g. Moodle). Thanks for your promotion. I'm waiting for an author to release a new version =)

Or you and your command are looking for new developers?😂

OliEfr commented:
@FurTactics Plenty of ideas and short of labour, as always:D

I am not at TUD anymore (munich now), and didn't know that [https://tud.uni-leipzig.de/moodle2/](https://tud.uni-leipzig.de/moodle2/) exists. That's an interesting domain 😅 True, tampermonkey scripts are more flexible and easier to configure. Seems also a bit more easy to integrate with FF-Android atm. Its also easy to integrate new content scripts in our [TUfast directory](https://github.com/TUfast-TUD/TUfast_TUD/tree/main/src/contentScripts) as well though. And we have cool stuff such as [forwarding](https://github.com/TUfast-TUD/TUfast_TUD/tree/main/src/contentScripts/forward) as well ;)

FurTactics committed:
Added commands to the Tampermonkey menu. Click statistics, dark theme in IDP and notifications.

Now in the Tampermonkey menu you can view click statistics, as well as erase all saved data. A blackout has been added to the IDP website. The configuration site has also been slightly updated and beautiful notifications are now sent.

FurTactics committed:
updated description of new features and added FAQ

spyfly commented:
Hi @FurTactics, thanks for your contribution.

This PR is not mergable in it's current state.

However I wouldn't mind adding 2FA support (Option #1) and the new features. Feel free to create a new PR where you add yourself in the `@author` field.

The 2FA Option 2 is a bit more complicated. You've included some copy pasted library code, but that would not be the way to go. It would require adding a build step for the script where an existing library is bundled together with the userscript.

FurTactics committed:
Uncomment line to press "submit" button

FurTactics committed:
Update because of renovation of IDP login page

Dark theme was changed

FurTactics committed:
Handle undefined values of credentials
